libs/prelude/Prelude/Maybe.idr

```diff
@@ -86,6 +86,12 @@ lowerMaybe (Just x) = x
 raiseToMaybe : (Monoid a, Eq a) => a -> Maybe a
 raiseToMaybe x = if x == neutral then Nothing else Just x
 
+||| Return a pair if both arguments are Just
+both : Maybe a -> Maybe b -> Maybe (a, b)
```

Comment:
This can be implemented as `both ma mb = [| MkPair ma mb |]`

Reply:
I suppose that renders this function useless.